types/coin.go

```diff
@@ -600,7 +600,7 @@ var (
 	// a letter, a number or a separator ('/').
 	reDnmString = `[a-zA-Z][a-zA-Z0-9/]{2,127}`
 	reAmt       = `[[:digit:]]+`
-	reDecAmt    = `[[:digit:]]*\.[[:digit:]]+`
+	reDecAmt    = `[[:digit:]]+(?:\.[[:digit:]]+)?|\.[[:digit:]]+`
```

Comment:
Can you explain what's the difference between the old one and the new one?

Reply:
The old one:
* `[[:digit:]]*`: optional digits followed by
* `\.[[:digit:]]+`: one dot `.` character followed by one or more digits

The new one:
* `[[:digit:]]+`: one or more digits followed by either of the following:
  * (?:\.[[:digit:]]+)?: optional named group starting with one dot `.` character and followed by 1 or more digits
  * `\.[[:digit:]]+`: one dot `.` character followed by 1 or more digits

I'll test this separately, but neither I can grasp why the new one works better?

Reply:
The old one don't parse `1`, the new one does. So with the new one, one can write both `1atom` and `0.1atom`.